ngx_http_akamai_g2o_module.c

```diff
@@ -308,8 +320,8 @@ check_has_g2o_headers(ngx_http_request_t *r, ngx_http_akamai_g2o_loc_conf_t  *al
     }
 
     // nonce is correct
-    if (ngx_strcmp(nonce.data, alcf->nonce.data)) {
-        ngx_log_error(alcf->log_level, r->connection->log, 0, "g2o nonce %V incorrect", &nonce);
+    if (ngx_strncmp(header_nonce.data, nonce.data, nonce.len)) {
```

Comment:
Better change to - if (nonce.len != header_nonce.len || ngx_memcmp(header_nonce.data, nonce.data, nonce.len))

Reply:
Is it better for performance or reliability? Just to understand the change.

Reply:
In my view, it's more of a pattern of how to compare two ngx_str_t's - first check the size, then compare the data.
When the two strings are usually different, it makes sense to check the size first and avoid reading the strings when the sizes do not match. In this case, 99.99% of the time the strings will be equal, so this claim does not apply. 
It is possible that memcmp can work somewhat faster, since it can work with dwords/qwords, but I doubt it will be noticeable.

Reply:
I couldn't find a lot of examples using ngx_memcmp to compare two ngx_str_t on nginx code but only two of them (ngx_string.c lines 1797 and 1840) without check the size first.
But ok I can change that piece of code to work that way.

Reply:
It is true that memcmp is not used much in nginx core, usually they use some form of strcmp, not sure why... but when comparing two ngx_str_t's (not ngx_str_t to a static string), there is a size check. 
For example - https://github.com/nginx/nginx/blob/master/src/http/ngx_http_variables.c#L418, there are ~4 other such conditions in this file.

Reply:
Done.